### DISCUSSION TITLE
webauthn support?

### DISCUSSION THREAD

**Discussion by tt0mmy** [Web3Auth/Web3Auth#487] [Help]
Does Web3Auth support Webauthn as a provider? Torus/OpenLogin seems to support it but from https://github.com/Web3Auth/Web3Auth/blob/master/packages/ui/src/config.ts it looks like it's not supported in Web3Auth.  If not, are there any suggestions on how to support it?

Although https://docs.tor.us/deprecated/customauth/api-reference/usage#gettoruskey docs is under the `deprecated` section...

**Comment by shahbaz17** [comment]
Yes, Web3Auth does support webauthn as an adapter, but it can be used only along with openloginAdapter.

Check out https://web3auth.io/docs/sdk/web/openlogin#loginsettings

**Reply by tt0mmy** [reply]
thanks for link! i'm now able to call `connectTo('webauthn')` and a new window loads but i'm seeing this after TouchID:
![image](https://user-images.githubusercontent.com/7282254/186993024-74b351cd-22ee-44f5-8dbc-824bd0e98e8f.png)

I am on Testnet with `chainNamespace: CHAIN_NAMESPACES.OTHER` if that makes a difference.

is it possible to use a hardware authenticator via `authenticatorAttachment: 'cross-platform'` to allow for yubikey, ledger, etc..., instead of FaceID/TouchID?

are there any docs/diagrams of web3auth in regards to web3authn provider flow?

**Comment by himanshuchawla009** [comment]
We don't support webauthn as an adapter for web3auth. It is something that can be used only along with openloginAdapter. 

Users can enable touchId/face from their accounts settings on https://app.openlogin.com.

**Reply by tt0mmy** [reply]
> We don't support webauthn as an adapter for web3auth. It is something that can be used only along with openloginAdapter.
> 
> Users can enable touchId/face from their accounts settings on https://app.openlogin.com.

I believe we have it configured properly to be used along with `openLoginAdapter`.  code snippet below:

```
        const web3authCore = new Web3AuthCore({
          chainConfig: {
            chainNamespace: CHAIN_NAMESPACES.OTHER,
          },
        })

        const openLoginAdapter = new OpenloginAdapter({
          adapterSettings: {
            network: "testnet",
            clientId,
          },
        })
        web3authCore.configureAdapter(openLoginAdapter)

        await web3authCore.init()

// usage.....

      await web3authCore?.connectTo(
        WALLET_ADAPTERS.OPENLOGIN,
        {
          loginProvider: 'webauthn',
        },
      )
```


> Users can enable touchId/face from their accounts settings on https://app.openlogin.com/.

I was trying to move away from FaceID/TouchID and use a hardware authenticator such as a Yubikey, Ledger, etc...
Is that possible with the `openLoginAdapter`/`web3auth`?

**Reply by yashovardhan** [reply]
Hey @tantommy 

Please have a look at our SDK reference for openlogin adapter. There is no loginProvider as webauthn available. The configuration code is wrong here.

https://web3auth.io/docs/sdk/web/openlogin

**Reply by yashovardhan** [reply]
Please note that on the first login, they will have to use any other provider like Google/ email passwordless etc. Consequently, a user can manually enable webauthn on further logins using app.openlogin.com.

If you want only a webauthn login, you can implement that on your own and pass over an JWT id_token to Web3Auth on successful authentication. You can also use services like Auth0, Firebase or AWS Cognito to enable this

https://web3auth.io/docs/custom-authentication/federated-identity-providers
https://web3auth.io/docs/custom-authentication/byo-jwt-providers

**Reply by tt0mmy** [reply]
> Hey @tantommy
> 
> Please have a look at our SDK reference for openlogin adapter. There is no loginProvider as webauthn available. The configuration code is wrong here.
> 
> https://web3auth.io/docs/sdk/web/openlogin

to clarify, do you mean `webauthn` from below is not valid?
![image](https://user-images.githubusercontent.com/7282254/189962349-ca8fe560-04ca-4f68-8969-cadf11ef6afb.png)
